salt/states/x509_v2.py

```diff
@@ -1578,8 +1579,24 @@ def _file_managed(name, test=None, **kwargs):
         raise SaltInvocationError("test param can only be None or True")
     # work around https://github.com/saltstack/salt/issues/62590
     test = test or __opts__["test"]
-    res = __salt__["state.single"]("file.managed", name, test=test, **kwargs)
-    return res[next(iter(res))]
+    file_managed = __states__["file.managed"]
```

Comment:
Afaik, `test` is a parameter for `state.apply` and is merged into `__opts__`, which is accessed directly by both the `file` execution and state module. Or does `__states__` somehow wrap returned functions to be able to override `test`?

Reply:
Ahh... you're right...
So, in any case we need to override `__opts__`. 

What about keeping that way, but simplify:
```python
if test not in [None, True]:
        raise SaltInvocationError("test param can only be None or True")
    # work around https://github.com/saltstack/salt/issues/62590
    opts = copy.deepcopy(__opts__)
    opts['test'] = test or __opts__["test"]
    
    file_managed = __states__["file.managed"]
    with salt.utils.context.func_globals_inject(
        file_managed, __opts__=opts):
            return file_managed(name, **kwargs)
```
Like, there is no need to execute check_perms, file_manage execution modules while in the end `file_managed` will do that for you, right?

Reply:
Tests summary:
```
tests/pytests/integration/states/test_x509_v2.py::test_certificate_managed_remote PASSED
tests/pytests/integration/states/test_x509_v2.py::test_certificate_managed_remote_no_changes[existing_cert0] PASSED
tests/pytests/integration/states/test_x509_v2.py::test_certificate_managed_remote_policy_change[existing_cert0] PASSED
tests/pytests/integration/states/test_x509_v2.py::test_certificate_managed_remote_ca_cert_change[existing_cert0] PASSED
tests/pytests/integration/states/test_x509_v2.py::test_certificate_managed_remote_no_changes_signing_policy_override[existing_cert0] PASSED
tests/pytests/integration/states/test_x509_v2.py::test_certificate_managed_remote_renew[existing_cert0] PASSED
tests/pytests/integration/states/test_x509_v2.py::test_privkey_new_with_prereq PASSED
tests/pytests/integration/states/test_x509_v2.py::test_privkey_new_with_prereq_pkcs12 PASSED
tests/pytests/integration/states/test_x509_v2.py::test_file_managed_does_not_run_in_test_mode_after_x509_v2_invocation_without_changes[existing_privkey0] PASSED
```

Reply:
The problem is that the file execution module still runs with the original, non-test opts in that case. In theory, patching one __salt__["file.*"] is enough though. I opted for the current approach for clarity and in case `func_globals_inject` at some point actually only patches one `func`, but removing one of the patch operations would work.

Reply:
Yeap....Funcs are still not passing.
Seems like that approach is the best one at the moment as we need to patch every callable function for the state.